protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -814,6 +815,38 @@ private long parseUnsignedInteger(String value) throws DecodingException {
         boolean isValid(char chr);
     }
 
+    /**
+     * Verifies subsequent characters match a specified string
+     */
+    public static class StringValidator implements CharacterValidator {
+        public static StringValidator caseIncentive(String expectedString) {
+            return new StringValidator(expectedString,
+                (c1, c2) -> isaBoolean(c1, c2));
+        }
+
+        public static boolean isaBoolean(Character c1, Character c2) {
+            return Character.toUpperCase(c1) == Character.toUpperCase(c2);
+        }
+
+        private final String expectedString;
+        private final BiPredicate<Character, Character> equalityTester;
+        private int position = 0;
+
+        private StringValidator(String expectedString, BiPredicate<Character, Character> equalityTester) {
+            this.expectedString = expectedString;
+            this.equalityTester = equalityTester;
+        }
+
+        @Override
+        public boolean isValid(char chr) {
```

Comment:
there's no way I can understand that isValid changes StringValidator internal state
Encapsulation here looks like a dangerous thing

Reply:
Would javadoc be enough or we need to be thinking of an alternative solution?